packages/ember-metal/lib/instrumentation.js

```diff
@@ -155,7 +155,7 @@ export function _instrumentStart(name, _payload) {
   var timestamp = time();
   for (i = 0; i < listeners.length; i++) {
     listener = listeners[i];
-    beforeValues[i] = listener.before(name, timestamp, payload);
+    beforeValues[i] = typeof listener.before === 'function' ? listener.before(name, timestamp, payload) : null;
```

Comment:
is the value every anything but a function? Can this simply check truthiness?

Making `before` optional would call for a documentation update as well.